src/components/watertracker/WaterTracker1L.java

```diff
@@ -0,0 +1,204 @@
+     * Logs water consumption from a specified source.
+     *
+     * @param source the name of the water source as a {@code String}
+     * 
+     * @throws AssertionError if {@code source} is null, empty, or not in the
+     * water sources map
+     * 
+     * @ensures <pre> if source is in DOMAIN(waterSources) then
+     * totalWaterConsumed = #totalWaterConsumed + waterSources[source] </pre>
+     */
+    @Override
+    public void drank(String source) {
+        assert source != null && !source
+                .isEmpty() : "Violation of: source is not null or empty";
+        assert this.waterSources.containsKey(
+                source) : "Violation of: source is in DOMAIN(waterSources)";
+
+        // Add the amount of water from source to the total consumed
+        double ounces = this.waterSources.get(source);
+        this.goalAmount -= ounces; // Reduce the goal amount by the consumed amount
+        System.out.println("Drank " + ounces + " oz from " + source);
+    }
+
+    /*
+     * Gets the map of water sources.
+     *
+     * @return the {@code Map} of water sources and their respective amounts
+     */
+    @Override
+    public Map<String, Double> getWaterSources() {
+        return this.waterSources;
+    }
+
+    /*
+     * Gets the name of the hydration goal.
+     *
+     * @return the name of the hydration goal as a {@code String}
+     */
+    @Override
+    public String getGoalName() {
+        return this.goalName;
+    }
+
+    /*
+     * Gets the target hydration goal amount.
+     *
+     * @return the hydration goal amount as a {@code double}
+     */
+    @Override
+    public double getGoalAmount() {
+        return this.goalAmount;
+    }
+
+    /*
+     * Sets the hydration goal.
+     *
+     * @param goalName the name of the hydration goal as a {@code String}
+     * 
+     * @param goalAmount the target hydration goal amount as a {@code double}
+     * 
+     * @throws AssertionError if {@code goalName} is null or empty or if {@code
+     * goalAmount} is negative
+     */
+    @Override
+    public void setGoal(String goalName, double goalAmount) {
+        if (goalName == null || goalName.isEmpty() || goalAmount < 0) {
+            throw new AssertionError("Invalid goal name or amount.");
+        }
+        this.goalName = goalName;
+        this.goalAmount = goalAmount;
+    }
+
+    /*
+     * Standard methods -------------------------------------------------------
+     */
+
+    /*
+     * Clears the water tracker by resetting the representation.
+     *
+     * @ensures <pre> waterSources = {} and goalName = "Default Goal" and
+     * goalAmount = 0.0 </pre>
+     */
+    @Override
+    public final void clear() {
+        this.createNewRep();
+    }
+
+    /*
+     * Creates a new instance of {@code WaterTracker1L}.
+     *
+     * @return a new instance of {@code WaterTracker1L}
+     */
+    @Override
+    public final WaterTracker1L newInstance() {
+        return new WaterTracker1L();
```

Comment:
This is probably fine, but there's a way we do it in most projects that is preferred.